plugins/glustershd/transactions.go

```diff
@@ -0,0 +1,115 @@
+package glustershd
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+
+	"github.com/gluster/glusterd2/glusterd2/brick"
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/glusterd2/servers/sunrpc/dict"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+)
+
+func getHxlChildrenCount(volinfo *volume.Volinfo) (int, string) {
+	if volinfo.Type == volume.Replicate || volinfo.Type == volume.DistReplicate {
+		return volinfo.Subvols[0].ReplicaCount, "replicate"
+	}
+	return volinfo.Subvols[0].DisperseCount, "disperse"
+}
+
+func addHxlatorToDict(reqDict map[string]string, volinfo *volume.Volinfo, index int, count int, xlType string) map[string]string {
+	key := fmt.Sprintf("xl-%d", count)
+	xName := fmt.Sprintf("%s-%s-%d", volinfo.Name, xlType, index)
+	reqDict[key] = xName
+	reqDict[xName] = strconv.Itoa(index)
+	return reqDict
+}
+
+func selectHxlatorsWithBricks(volinfo *volume.Volinfo, healType string) map[string]string {
```

Comment:
AFR folks would be the best people to review this and related functions

Reply:
@prashanthpai  I have requested @itisravi to review index heal API again.